CitationStyleLanguagePlugin.inc.php

```diff
@@ -428,10 +482,10 @@ public function getCitation($request, $article, $citationStyle = 'apa', $issue =
 
         // DOI
         if ($issue && $issue->getPublished()) {
-            $citationData->DOI = $article->getStoredPubId('doi');
+            $citationData->DOI = $submission->getStoredPubId('doi');
```

Comment:
In 3.4, we've moved DOIs into the core application. Get the DOI from the publication now:

```php
$citationData->DOI = $publication->getDoi();
```